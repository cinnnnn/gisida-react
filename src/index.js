import React from 'react'
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import App from './components/App/App';
import registerServiceWorker from './registerServiceWorker';
import MapContainer from './containers/MapContainer';
import MenuContainer from './containers/MenuContainer'
import StyleSelectorContainer from './containers/StyleSelectorContainer';


import Menu from './components/Menu/Menu';

import { initStore } from 'gisida';

const store = initStore();

ReactDOM.render(
  <Provider store={store}>
    <App>
      <MapContainer />
      <MenuContainer />
      <StyleSelectorContainer />
    </App>
  </Provider>,
  document.getElementById('root'));
registerServiceWorker();
