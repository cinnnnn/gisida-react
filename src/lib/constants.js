export const NUTRITION_SITES_LIVE = 'nutrition-sites-live';
export const NUTRITION_SITES_LIVE_HIGHLIGHT = 'nutrition-sites-live-highlight';
export const FACILITY_ID = 'facility_id';
export const ICON_OPACITY = 'icon-opacity';
export const ALL = 'all';