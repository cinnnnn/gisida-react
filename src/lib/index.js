import App from './components/App/App';
import MenuItems from './components/MenuItems/MenuItems';
import Map from './components/Map/Map';
import StyleSelector from './components/StyleSelector/StyleSelector';
import Legend from './components/Legend/Legend';

export {
 App, MenuItems, Map, StyleSelector, Legend
};

