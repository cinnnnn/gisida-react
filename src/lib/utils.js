export function formatNum(num, decimal) {
  let x = (`${num}`).length;
  if (Number.isInteger(num) && x > 3) {
    const pow = 10 ** decimal;
    x -= x % 3;
    return Math.round(num * pow / (10 ** x)) / pow + ' kMGTPE'[x / 3];
  }
  return num;
}

export function getLastIndex(arr, item) {
  const indices = [];

  for (let i = 0; i < arr.length; i += 1) {
    if (arr[i] === item) { indices.push(i); }
  }
  return indices[indices.length - 1];
}


export function groupBy(collection, property) {
  var i = 0, val, index,
    values = [], result = [];
  for (; i < collection.length; i++) {
    val = collection[i][property];
    index = values.indexOf(val);
    if (index > -1)
      result[index].push(collection[i]);
    else {
      values.push(val);
      result.push([collection[i]]);
    }
  }
  return result;
}

export function detectIE() {
  var ua = window.navigator.userAgent;

  // IE 10
  // ua = 'Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.2; Trident/6.0)';

  // IE 11
  // ua = 'Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko';

  var msie = ua.indexOf('MSIE ');
  if (msie > 0) {
    // IE 10 or older => return true
    return true;
  }

  var trident = ua.indexOf('Trident/');
  if (trident > 0) {
    // IE 11 => return true
    return true;
  }
  // other browser
  return false;
}