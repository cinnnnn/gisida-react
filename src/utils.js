export function formatNum(num, decimal) {
  let x = (`${num}`).length;
  if (Number.isInteger(num) && x > 3) {
    const pow = 10 ** decimal;
    x -= x % 3;
    return Math.round(num * pow / (10 ** x)) / pow + ' kMGTPE'[x / 3];
  }
  return num;
}

export function getLastIndex(arr, item) {
  const indices = [];

  for (let i = 0; i < arr.length; i += 1) {
    if (arr[i] === item) { indices.push(i); }
  }
  return indices[indices.length - 1];
}


export function groupBy(collection, property) {
  var i = 0, val, index,
    values = [], result = [];
  for (; i < collection.length; i++) {
    val = collection[i][property];
    index = values.indexOf(val);
    if (index > -1)
      result[index].push(collection[i]);
    else {
      values.push(val);
      result.push([collection[i]]);
    }
  }
  return result;
}