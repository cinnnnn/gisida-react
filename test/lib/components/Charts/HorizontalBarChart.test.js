import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { HorizontalBarChart } from '../../../../src/lib';
import {wrapper} from '../../../fixtures/chartComponentWrapper'

const props = {
	series: [],
    categories: []
}

const mountWrapper = wrapper(HorizontalBarChart, mount, props);

describe('HorizontalBarChart', () => {
	it('HorizontalBarChart component renders correctly', () => {
		const json = toJson(mountWrapper)
		expect(json).toMatchSnapshot();
	})

	it('should set state', () => {
		mountWrapper.setState({ show: true });
		expect(mountWrapper.state('show')).toEqual(true)
	})

	it('should set props', () => {
		const propsUpdateMock = jest.spyOn(mountWrapper.instance(), 'componentWillReceiveProps');
		mountWrapper.setProps({ name: 'John Doe' });
		expect(mountWrapper.props().name).toEqual('John Doe');
		expect('series' in mountWrapper.props()).toEqual(true);
		expect(propsUpdateMock).toHaveBeenCalledTimes(1);
	})

	it('should unmount', () => {
		const componentUnmountMock = jest.spyOn(mountWrapper.instance(), 'componentWillUnmount');
		mountWrapper.unmount()
		expect(componentUnmountMock).toHaveBeenCalledTimes(1);
	})

});