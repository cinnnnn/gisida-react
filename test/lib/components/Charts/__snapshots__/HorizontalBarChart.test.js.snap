// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalBarChart HorizontalBarChart component renders correctly 1`] = `
<HorizontalBarChart
  categories={Array []}
  series={Array []}
>
  <div />
</HorizontalBarChart>
`;
